### PR TITLE
Remove email validation from js component

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Remove email validation from js component

#### Why?

To be able to display proper error messages and prevent form saving

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/MemoryFormStore.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import {action, observable} from 'mobx';
 import type {IObservableValue} from 'mobx';
 import jsonpointer from 'json-pointer';
-import {createAjv} from '../../../utils';
+import {createAjv} from '../../../utils/Ajv';
```
Comment (luca-rath) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/MemoryFormStore.js]: @nnatter Do you have an idea why this change fixes this error? https://github.com/sulu/sulu/pull/5815/checks?check_run_id=1880372111

Comment (niklasnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/MemoryFormStore.js]: because these test classes mock the `sulu-admin-bundle/utils` modules instead of the specific module (for example `sulu-admin-bundle/utils/Translator`) 🙈  
see for example https://github.com/sulu/sulu/blob/466edcb9beb7debe5c64e75bade7d04bbb6164de/src/Sulu/Bundle/MediaBundle/Resources/js/containers/MediaCollection/tests/MediaCollection.test.js#L191-L193

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/MemoryFormStore.js]: 🙈

File: src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// @flow
+import IsEmail from 'isemail';
+import {FormatValidator} from 'ajv';
+
+const idnEmailValidator: FormatValidator = (data: string): boolean => {
```
Comment (niklasnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js:5]: would prefer to update `ajv` and add the `ajv-formats` and `ajv-formats-draft2019` package. just had a look at the changelog, i think the update should not cause problems 🙂

File: src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/StringMetadata.php [hunk 3]
```diff
@@ -53,6 +63,10 @@ public function toJsonSchema(): array
             $jsonSchema['pattern'] = $this->pattern;
         }
 
+        if (null !== $this->format) {
+            $jsonSchema['format'] = $this->format;
```
Comment (niklasnatter) [inline src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/StringMetadata.php:67]: this is not tested, right?

File: src/Sulu/Bundle/PageBundle/Content/Types/Email.php [hunk 4]
```diff
@@ -11,15 +11,48 @@
...
  */
-class Email extends SimpleContentType
+class Email extends SimpleContentType implements PropertyMetadataMapperInterface
 {
     public function __construct()
     {
         parent::__construct('Email', '');
     }
+
+    public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata): PropertyMetadata
```
Comment (niklasnatter) [inline src/Sulu/Bundle/PageBundle/Content/Types/Email.php:33]: not sure if we tested this method for other `PropertyMetadataMapper`s

File: src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js (hunk 2 above)
Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js:5]: I would prefer not to install `ajv-formats-draft2019`, because it contains formats which are not working as expected, see https://github.com/sulu/sulu/issues/5681#issuecomment-777442172 ... So we still need to implement the `idn-email` format by ourselves, but it's not that complicated with using `isemail`, so that should be fine

File: src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/StringMetadata.php (hunk 3 above)
Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/StringMetadata.php:67]: The whole class is not tested, but I'll write a test for it

File: src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js (hunk 2 above)
Comment (niklasnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/formats/idnEmailValidator.js:5]: just saw that the `ajv-formats-draft2019` is a community package. is okay for me too 🙂

File: src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// @flow
+import type {KeywordDefinition} from 'ajv';
+
+const nameDefinition: KeywordDefinition = {
```
Comment (niklasnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js:4]: not really sure what this does? can you give me a hint? 😇

**Review by niklasnatter — APPROVED** [review]
looks good otherwise!

File: src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js (hunk 5 above)
Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js:4]: The json schema returned by our api often looks like this

~~~json
{
  "type": "object",
  "properties": {
    "foo": {
      "name": "foo",
      "type": "string"
    }
  }
}
~~~

But `name` is not a valid json schema keyword, so we need to explicitly allow it.

Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js:4]: In ajv v7, the strict mode is enabled by default, therefore this wasn't needed before. We could also disable that, but I think it's generally good to have strict mode enabled

Comment (niklasnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/utils/Ajv/keywords/nameDefinition.js:4]: alright sounds good 👍
